lib/ddtrace/profiling/transport/http.rb

```diff
@@ -65,20 +65,21 @@ def self.default_headers
         end
 
         private_class_method def self.default_adapter
-          :net_http
+          Datadog::Ext::Transport::HTTP::ADAPTER
         end
```

Comment:
I'll give this a try.

Reply:
Hopefully I clarified above what I meant and why I believe this can be removed :)

Reply:
Now that I refactor all adapters to use agent settings, I'm actually confused in a different way on how to remove this method.

This `default_adapter` is only used in `configure_for_agentless`, and `configure_for_agentless` does not build its settings using a `AgentSettings`, likely because `AgentSettings` currently is only built from a `Datadog::Configuration::Settings`-like object and not a `site_uri`.

Are you suggesting I add a new data source for `AgentSettings`, based on the profiler's `site_uri`?

Reply:
> Are you suggesting I add a new data source for AgentSettings, based on the profiler's site_uri?

Ah, no. For the `configure_for_agent` case, we can use the `AgentSettings` as you pointed out. For the `configure_for_agentless`, it doesn't make sense for agentless to be anything other than "talk to the backend directly via http", otherwise, and thus we can effectively hardcode it to use the http adapter (e.g. inline the constant in that method).

The agentless mode is always going to be a bit weird; and I don't think it's worth trying to fit it into the `AgentSettings` object model since it's a profiler-specific thing.